lib/types/thermostat.js

```diff
@@ -0,0 +1,345 @@
+var inherits = require('util').inherits;
+
+module.exports = function (HAPnode, config, functions) {
+    var Accessory = HAPnode.Accessory;
+    var Service = HAPnode.Service;
+    var Characteristic = HAPnode.Characteristic;
+    var uuid = HAPnode.uuid;
+    var debug = HAPnode.debug;
+
+    var module = {};
+
+    module.newDevice = function (device, tempDisplayUnit) {
+        var temperatureDisplayUnit = tempDisplayUnit;
+        var Thermostat = {
+            device_url: function(request_type){
+                host = 'http://'+config.veraIP+':3480/data_request'
+                return host+'?id='+request_type+'&DeviceNum='+device.id
+            },
+
+            remoteRequest: function(url, params, callback){
+                debug("Requesting: %s", url);
+                //HAPnode.request.debug = true
+                return HAPnode.request({
+                    method:'GET',
+                    uri: url,
+                    qs: params,
+                    resolveWithFullResponse: true
+                }).then(callback.bind(this)).catch(function(e){
+                    debug(e.error);
+                    debug(e.options);
+                    debug(e.response);
+                });
+            },
+
+            getVeraVariable: function(serviceId, variable, callback){
+                var url = this.device_url('variableget');
+                var params = {serviceId: serviceId, Variable: variable}
+                return this.remoteRequest(url, params, callback);
+            },
+
+            executeVeraAction: function(params){
+                var url = this.device_url('action');
+                var callback = function(){};
+                return this.remoteRequest(url, params, callback);
+            },
+
+            getCurrentHeatingCoolingState: function(){
+                var serviceId = 'urn:upnp-org:serviceId:HVAC_UserOperatingMode1';
+                var variable = 'ModeStatus';
+                var callback = function (response){
+                    if (response.statusCode === 200){
+                        var vera_state = response.body;
+                        var state;
+                        switch(vera_state){
```

Comment:
@tzonto makes sense, although from an iOS UI standpoint it means that a thermostat that is idling still shows as either orange or blue, rather than gray (not running). I'll play around with it, as I have a Nest using the WWN plugin and a Z-Wave thermostat on the porch to control a gas fireplace. Ideally they should both behave the same way.